arangod/IResearch/IResearchViewSingleServer.cpp

```diff
@@ -0,0 +1,111 @@
+//////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2017 EMC Corporation
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is EMC Corporation
+///
+/// @author Lars Maier
+////////////////////////////////////////////////////////////////////////////////
+
+#include "IResearchCommon.h"
+#include "IResearchDocument.h"
+#include "IResearchFeature.h"
+#include "IResearchFilterFactory.h"
+#include "IResearchLink.h"
+#include "IResearchLinkHelper.h"
+
+#include "StorageEngine/TransactionState.h"
+#include "StorageEngine/StorageEngine.h"
+#include "RestServer/DatabaseFeature.h"
+#include "RestServer/DatabasePathFeature.h"
+#include "RestServer/FlushFeature.h"
+#include "Transaction/Methods.h"
+#include "Transaction/StandaloneContext.h"
+#include "Utils/ExecContext.h"
+#include "velocypack/Iterator.h"
+#include "VocBase/LogicalCollection.h"
+#include "VocBase/vocbase.h"
+#include "VocBase/LogicalView.h"
+#include "Logger/LogMacros.h"
+
+#include "IResearchView.h"
+#include "IResearchViewSingleServer.h"
+
+namespace arangodb {
+namespace iresearch {
+
+/*static*/ std::shared_ptr<LogicalView> IResearchViewSingleServer::make(
```

Comment:
why can the code from IResearchView be reused?

Reply:
Because thats what happens before in the single server case. The only reason I did this, is that the DBServer version reuses the code of the single server.

Reply:
I am not sure I follow why there needs to be code duplication and the necessity for this class instead of just using IResearchView::make(...)